submission/persistence-filesystem/src/main/kotlin/ac/uk/ebi/biostd/persistence/filesystem/fire/FireService.kt

```diff
@@ -29,21 +33,47 @@ class FireService(
      * different path) and if so, even if the file exists in FIRE, it gets duplicated to ensure consistency.
      */
     fun getOrPersist(sub: ExtSubmission, file: ExtFile): FirePersistResult {
-        val expectedPath = "/${sub.relPath}/${file.relPath}"
         return when (file) {
-            is FireFile -> fromFireFile(file, expectedPath)
-            is NfsFile -> when (file.type) {
-                FILE -> fromNfsFile(file, expectedPath)
-                DIR -> fromNfsFile(file.copy(file = compress(sub, file.file)), "$expectedPath.zip")
+            is FireFile -> {
+                val downloadFile = { client.downloadByFireId(file.fireId, file.fileName) }
+                reuseOrPersistFireFile(file, "/${sub.relPath}/${file.relPath}", downloadFile)
+            }
+            is NfsFile -> {
+                val nfsFile = if (file.type == FILE) file else extDirectory(sub, file)
+                val downloadFile = { nfsFile.file }
+                return reuseOrPersistFireFile(nfsFile, "/${sub.relPath}/${nfsFile.relPath}", downloadFile)
             }
         }
     }
 
-    private fun fromNfsFile(file: NfsFile, expectedPath: String): FirePersistResult =
-        reuseOrPersistFireFile(file, expectedPath) { file.file }
+    private fun extDirectory(sub: ExtSubmission, directory: NfsFile): NfsFile {
+        fun compress(file: File): File {
+            val tempFolder = fireTempDirPath.resolve("${sub.accNo}/${sub.version}")
+            tempFolder.mkdirs()
+
+            val target = tempFolder.resolve("${file.name}.zip")
+            Files.deleteIfExists(target.toPath())
+            ZipUtil.pack(file, target, true)
+            return target
+        }
+
+        fun calculateMd5(dir: File): String {
```

Comment:
Since we're always compressing the directory before uploading it to fire, why do we need to calculate the md5 in this way? Or is this the same value as the compressed directory?

I'm just asking because I'm not sure fire will take this md5 for the upload as valid and we'll need to recalculate it anyway and also, if we're trying to find it by md5 in order to reuse it, I'm not sure if this md5 will do